apps/settings/js/modules/simpin.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * This is a simple model that can help us mimic the status change event
+ * when enabling / disabling simcard.
+ *
+ * @module SimPin
+ */
+define(function(require) {
+  'use strict';
+
+  var Module = require('modules/base/module');
+  var SettingsUtils = require('modules/settings_utils');
+  var EventEmitter = require('modules/base/event_emitter');
+
+  const EVENT_NAME = 'simpin-';
+  const APIS = [
+    'unlockCardLock',
+    'setCardLock',
+    'updateContact',
+    'getCardLock',
+    'getCardLockRetryCount'
+  ];
+
+  var SimPin = Module.create(function() {
+    var supportedEvents = [];
+    APIS.forEach((apiName) => {
+      supportedEvents.push(EVENT_NAME + apiName + '-success');
+      supportedEvents.push(EVENT_NAME + apiName + '-fail');
+    });
+    this.super(EventEmitter).call(this, supportedEvents);
+  }).extend(EventEmitter);
+
+  APIS.forEach(function(apiName) {
+    SimPin.prototype[apiName] = function(cardIndex) {
+      if (arguments.length < 2) {
+        throw new Error('You are using SimPin API wrongly and you may forget' +
+          'to put your cardIndex, please check again');
+      }
+
+      var self = this;
+      var passedArguments = [].slice.call(arguments, 1);
+      var icc = SettingsUtils.getIccByCardIndex(cardIndex);
```

Comment:
This function looks quite general and may be used in the future for other panels, right ?